components/ai_chat/content/browser/brave_search_responses.idl

```diff
@@ -0,0 +1,19 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+namespace brave_search_responses {
+  dictionary Answer {
+    DOMString text;
+  };
+
+  dictionary QuerySummary {
+    Answer[] answer;
+    DOMString query;
+  };
+
+  dictionary QuerySummaryResponse {
+    QuerySummary[] conversation;
+  };
```

Comment:
Do these structs correspond to the results from the Brave Search API?

Reply:
Yes, but I don't think we have API doc atm.
I listed a sample response here https://github.com/brave/reviews/issues/1732.